src/main/java/spleetwaise/transaction/ui/RightPanel.java

```diff
@@ -85,25 +96,58 @@ public RightPanel(CommandBox cb) {
         filterBtn.addEventHandler(MouseEvent.MOUSE_CLICKED, this::showFilterMenu);
     }
 
+    /**
+     * Updates the balance labels based on whether trackUndoneBalanceOnlyCheckBox is checked.
+     */
+    private void updateBalances() {
+        updateBalances(trackUndoneBalanceOnlyCheckBox.isSelected());
+    }
+
     /**
      * Updates the balance labels based on the current transactions.
      */
-    public void updateBalances() {
+    private void updateBalances(boolean isTrackingUndoneBalanceOnly) {
         ObservableList<Transaction> txns = CommonModelManager.getInstance().getFilteredTransactionList();
+
+        // Create predicates for each balance type (owe and owed)
+        FilterCommandPredicate youOwnFilter = createBalanceFilter(isTrackingUndoneBalanceOnly, NEGATIVE_SIGN);
+        FilterCommandPredicate ownYouFilter = createBalanceFilter(isTrackingUndoneBalanceOnly, POSITIVE_SIGN);
+
         youOwnLabel.setText(
-                "You Owe $" + calculateBalance(txns, new AmountSignFilterPredicate(NEGATIVE_SIGN)).toString());
+                "You Owe $" + calculateBalance(txns, youOwnFilter).toString());
         ownYouLabel.setText(
-                "You are Owed $" + calculateBalance(txns, new AmountSignFilterPredicate(POSITIVE_SIGN)).toString());
+                "You are Owed $" + calculateBalance(txns, ownYouFilter).toString());
```

Comment:
typoe own?

Reply:
I can change variable name to `youOweFilter` and `youAreOwedFilter` for clarity